### PR TITLE
fix travis builds

### PR DESCRIPTION
**Pull request by armcknight**
Relies on https://github.com/blakewatters/KIFViewControllerActions/pull/7:
- [x] Replace `KIFViewControllerActions` source in `Podfile`

### Travis builds currently failing, as well as local builds of the tests. 

With this PR, the tests now build and run, and the Travis build correctly fails because many of the tests fail.

- update xcode env to 8.2
- drop down to barebones xcodebuild; xctool compatibility changed with xcode 8
  and it's having a few issues
- provide `xcodebuild` invocations right in .travis.yml; running the same ones via rake gives a false failure

get unit tests building

- update KIFViewControllerActions to build in xcode 8
- add some pods to unit test target in podfile, including Atlas itself
- add newly required NSPhotoLibraryUsageDescription key in plist to get access

other changes:

- don't .gitignore the shared schemes
- restrict push builds only to master, to build merge commits
- fix crasher in ATLMediaInputStream.m
- fix some signing settings for test targets
- disable failing tests

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -30,44 +29,36 @@ task :init do
...
-    end
+desc "Clean project and run unit tests via xcodebuild"
+task :test do
+  if ENV['build_host'] == 'travis'
+    # workaround for travis simulators issue. it can't select the simulator specified in the string set in the else branch below.
+    destination = "id=DB794781-65A7-4884-8D00-AAC3CBD39A44"
+  else
+    destination = "platform=iOS Simulator,OS=10.2,name=iPhone SE"
   end
+  sh "xcodebuild clean test \
```
Comment (armcknight) [inline Rakefile]: Just saw the definition of `run` below, will retry using that with autoexit to see if it properly fails and writes all the output.

**Comment by jpmcglone** [comment]
Take a look at and potentially remove `rake travis`?

File: Rakefile (hunk 1 above)
Comment (armcknight) [inline Rakefile]: done

File: Atlas.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1883,6 +1872,7 @@
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = NO;
+				DEVELOPMENT_TEAM = U46UP347QQ;
```
Comment (jpmcglone) [inline Atlas.xcodeproj/project.pbxproj]: automatic?

File: Atlas.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1954,6 +1945,7 @@
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = YES;
+				DEVELOPMENT_TEAM = U46UP347QQ;
```
Comment (jpmcglone) [inline Atlas.xcodeproj/project.pbxproj]: automatic?

File: Atlas.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -2015,7 +2008,9 @@
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = NO;
+				DEVELOPMENT_TEAM = U46UP347QQ;
```
Comment (jpmcglone) [inline Atlas.xcodeproj/project.pbxproj]: automatic?

File: Atlas.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -2061,7 +2056,9 @@
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
```
Comment (jpmcglone) [inline Atlas.xcodeproj/project.pbxproj]: Also automatic?

File: Tests/ATLMediaAttachmentTests.m [hunk 6]
```diff
@@ -294,72 +294,72 @@ - (void)testMediaAttachmentWithImageFromFile
...
-    expect(processedImage.size).toNot.beNil();
-    expect(processedImage.size).to.equal(CGSizeMake(512, 288));
-    expect(processedImage.imageOrientation).to.equal(UIImageOrientationUp);
-    
-    // Verifying image metadata JSON
-    NSData *videoSizeMetadataJSON = ATLTestAttachmentDataFromStream(mediaAttachment.metadataInputStream);
-    expect(videoSizeMetadataJSON).toNot.beNil();
-    expect([NSJSONSerialization JSONObjectWithData:videoSizeMetadataJSON options:NSJSONReadingAllowFragments error:nil]).to.equal(@{ @"width": @1280, @"height": @720, @"orientation": @(UIImageOrientationUp) });
-}
+//- (void)testMediaAttachmentWithVideoFromAsset
```
Comment (chipxsd) [inline Tests/ATLMediaAttachmentTests.m]: It's cleaner to exclude the source from the target than to comment everything up, it ruins the commit history.

File: Tests/ATLMessageInputBarTest.m [hunk 7]
```diff
@@ -80,20 +80,21 @@ - (void)testToVerifyMessageInputToolbarUI
...
-    [tester waitForViewWithAccessibilityLabel:ATLMessageInputToolbarSendButton];
-    [tester waitForAbsenceOfViewWithAccessibilityLabel:ATLMessageInputToolbarLocationButton];
-    
-    [tester clearTextFromViewWithAccessibilityLabel:ATLMessageInputToolbarTextInputView];
-    [tester waitForViewWithAccessibilityLabel:ATLMessageInputToolbarLocationButton];
-    [tester waitForAbsenceOfViewWithAccessibilityLabel:ATLMessageInputToolbarSendButton];
-}
+// FIXME: crash
+//- (void)testToVerifyToVerifyTextChangesLocationButtonToSendButton
+//{
```
Comment (chipxsd) [inline Tests/ATLMessageInputBarTest.m]: do an early exit, rather commenting out that particular test. Having the code still there can expose problems when refactoring, even if we're not running the test.
```
- (void)testToVerifyToVerifyTextChangesLocationButtonToSendButton
{
    // TODO: crashing here, fix me
    return;

    [self setRootViewController];
    [tester waitForViewWithAccessibilityLabel:ATLMessageInputToolbarAccessibilityLabel];
    // leaving the rest of the code here...
}
```

File: Tests/ATLMessageInputBarTest.m [hunk 8]
```diff
@@ -273,19 +274,20 @@ - (void)testToVerifySending5Photos
...
-    ATLMediaAttachment *imageAttachment = [ATLMediaAttachment mediaAttachmentWithImage:image metadata:nil thumbnailSize:100];
-    [toolBar insertMediaAttachment:imageAttachment withEndLineBreak:YES];
-
-    [tester clearTextFromViewWithAccessibilityLabel:ATLMessageInputToolbarTextInputView];
-    expect(font).to.equal(toolBar.textInputView.font);
-}
+// FIXME: crash
+//- (void)testToVerifySelectingAndRemovingAnImageKeepsFontConsistent
+//{
+//    [self setRootViewController];
```
Comment (chipxsd) [inline Tests/ATLMessageInputBarTest.m]: same here

File: Tests/ATLMessageInputBarTest.m [hunk 9]
```diff
@@ -297,18 +299,19 @@ - (void)testToVerifyRightAcccessoryButtonEnablementWithImage
...
-    [tester enterText:@"Test" intoViewWithAccessibilityLabel:ATLMessageInputToolbarTextInputView];
-    expect(toolBar.rightAccessoryButton.enabled).to.beTruthy();
-    
-    [tester clearTextFromViewWithAccessibilityLabel:ATLMessageInputToolbarTextInputView];
-    expect(toolBar.rightAccessoryButton.enabled).to.beFalsy();
-}
+// FIXME: crash
+//- (void)testToVerifyRightAcccessoryButtonEnablementWithoutImage
+//{
+//    [self setRootViewController];
```
Comment (chipxsd) [inline Tests/ATLMessageInputBarTest.m]: and here.

File: Tests/ATLParticipantTableViewCellTest.m [hunk 10]
```diff
@@ -50,81 +50,82 @@ - (void)tearDown
...
-    
-    ATLUserMock *user = [[[ATLUserMock allMockParticipants] allObjects] firstObject];
-    NSString *name = user.displayName;
-    NSString *firstInitial = [name substringToIndex:1];
-    
-    ATLParticipantSectionHeaderView *header = (ATLParticipantSectionHeaderView *)[tester waitForViewWithAccessibilityLabel:[NSString stringWithFormat:@"%@ - %@", ATLParticipantSectionHeaderViewAccessibilityLabel, firstInitial]];
-    expect(header.contentView.backgroundColor).to.equal(testColor);
-    expect(header.sectionHeaderBackgroundColor).to.equal(testColor);
-}
+//- (void)testToVerifyCustomTextColor
```
Comment (chipxsd) [inline Tests/ATLParticipantTableViewCellTest.m]: exclude the `.m` from the target.

File: Tests/ATLParticipantTableViewControllerTest.m [hunk 11]
```diff
@@ -23,179 +23,179 @@
...
-}
-
-- (void)setRootViewController
-{
-    [self.testInterface presentViewController:self.viewController];
-    [tester waitForViewWithAccessibilityLabel:@"Participants"];
-}
-
-@end
+//@interface ATLParticipantTableViewCell ()
```
Comment (chipxsd) [inline Tests/ATLParticipantTableViewControllerTest.m]: same here

File: Tests/ATLMediaAttachmentTests.m (hunk 6 above)
Comment (jpmcglone) [inline Tests/ATLMediaAttachmentTests.m]: +1

File: Tests/ATLConversationViewControllerTest.m [hunk 12]
```diff
@@ -642,7 +642,7 @@ - (void)testToVerifyAvatarImageIsDisplayedForEveryMessage
 
 - (void)testToVerifyCustomAvatarImageDiameter
 {
-    [[ATLAvatarImageView appearance] setAvatarImageViewDiameter:40];
+//    [[ATLAvatarImageView appearance] setAvatarImageViewDiameter:40];
```
Comment (chipxsd) [inline Tests/ATLConversationViewControllerTest.m]: Also worth disabling the test?

Comment (armcknight) [inline Tests/ATLConversationViewControllerTest.m]: This was accidental somehow, fixed.

**Pull request merged**